usermods/sht/usermod_sht.h

```diff
@@ -0,0 +1,312 @@
+    }
+
+    void publishTemperatureAndHumidityViaMqtt() {
+      if (!WLED_MQTT_CONNECTED) return;
+      char buf[128];
+
+      sprintf_P(buf, PSTR("%s/temperature"), mqttDeviceTopic);
+      mqtt->publish(buf, 0, false, String(shtCurrentTemp).c_str());
+      sprintf_P(buf, PSTR("%s/humidity"), mqttDeviceTopic);
+      mqtt->publish(buf, 0, false, String(shtCurrentHumidity).c_str());
+    }
+
+    void publishHomeAssistantAutodiscovery() {
+      if (!WLED_MQTT_CONNECTED) return;
+
+      char json_str[1024], buf[128];
+      size_t payload_size;
+      StaticJsonDocument<1024> json;
+
+      sprintf_P(buf, PSTR("%s Temperature"), serverDescription);
+      json[F("name")] = buf;
+      sprintf_P(buf, PSTR("%s/temperature"), mqttDeviceTopic);
+      json[F("stat_t")] = buf;
+      json[F("dev_cla")] = F("temperature");
+      json[F("stat_cla")] = F("measurement");
+      sprintf_P(buf, PSTR("%s-temperature"), escapedMac.c_str());
+      json[F("uniq_id")] = buf;
+      json[F("unit_of_meas")] = F("°C");
+      appendDeviceToMqttDiscoveryMessage(json);
+      payload_size = serializeJson(json, json_str);
+      sprintf_P(buf, PSTR("homeassistant/sensor/%s/%s-temperature/config"), escapedMac.c_str(), escapedMac.c_str());
+      mqtt->publish(buf, 0, true, json_str, payload_size);
+
+      json.clear();
+
+      sprintf_P(buf, PSTR("%s Humidity"), serverDescription);
+      json[F("name")] = buf;
+      sprintf_P(buf, PSTR("%s/humidity"), mqttDeviceTopic);
+      json[F("stat_t")] = buf;
+      json[F("dev_cla")] = F("humidity");
+      json[F("stat_cla")] = F("measurement");
+      sprintf_P(buf, PSTR("%s-humidity"), escapedMac.c_str());
+      json[F("uniq_id")] = buf;
+      json[F("unit_of_meas")] = F("%");
+      appendDeviceToMqttDiscoveryMessage(json);
+      payload_size = serializeJson(json, json_str);
+      sprintf_P(buf, PSTR("homeassistant/sensor/%s/%s-humidity/config"), escapedMac.c_str(), escapedMac.c_str());
+      mqtt->publish(buf, 0, true, json_str, payload_size);
+    }
+
+    void appendDeviceToMqttDiscoveryMessage(JsonDocument& root) {
+      JsonObject device = root.createNestedObject("dev");
+      device["ids"] = escapedMac.c_str();
+      device["name"] = serverDescription;
+      device["sw"] = versionString;
+      device["mdl"] = ESP.getChipModel();
+      device["mf"] = "espressif";
+    }
+
+    /*
+      * getId() allows you to optionally give your V2 usermod an unique ID (please define it in const.h!).
+      * This could be used in the future for the system to determine whether your usermod is installed.
+      */
+    uint16_t getId()
+    {
+      return USERMOD_ID_SHT;
+    }
+};
+
+// strings to reduce flash memory usage (used more than twice)
+// Config settings
+const char ShtUsermod::_name[]    PROGMEM = "SHT Temperature & Humidity Sensor";
+const char ShtUsermod::_enabled[] PROGMEM = "Enabled";
+const char ShtUsermod::_haMqttDiscovery[] PROGMEM = "Add-To-Home-Assistant-MQTT-Discovery";
```

Comment:
Same as above.
There is no need to capitalize first letters as it is done automatically.
I will add dash (-) to space handling there as well.